modules/dists/BlockDist.chpl

```diff
@@ -1624,3 +1641,21 @@ where useBulkTransferDist {
 
   return true;
 }
+
+proc newBlockDom(dom: domain) {
+  return dom dmapped Block(dom);
+}
```

Comment:
I wonder if we should do something special if `dom` is already Block-distributed. Or maybe even in all cases where `dom` is not DefaultRectangular. Ditto for all the analogous cases in this PR.

OK to merge as-is for now.

Reply:
I'm not sure.  If we did something like use `dom` directly in that case it would potentially cause pretty different behavior for things that look like the same index set.  The bounding box or locales used could be different which might be confusing.  OTOH maybe a person passing an already distributed domain really wanted that.

Reply:
Right. Maybe issue a warning to draw user attention, like "did you really mean that?".

Let's wait to hear from users on this.